### PR TITLE
Refs [TASK][26280] Handle add new Task event

### PR DESCRIPTION
**Pull request by quockhanhng**
## Related Tickets
- [#26280](https://edu-redmine.sun-asterisk.vn/issues/26280)

## WHAT
- Handle add new Task event

## HOW
- Pick a color tag, select due date(optional)
- Save new task to the database

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Reviewer1 (KhanhNQ) | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/getitdone/ui/newTask/NewTaskFragment.kt [hunk 1]
```diff
@@ -20,24 +43,91 @@ class NewTaskFragment : BaseBottomSheetFragment<NewTaskContract.View, NewTaskPre
...
+    private fun initListener() {
+        for (item in linearCardView.children) {
+            if (item is CardView) item.setOnClickListener { onColorPicker(item.tag as String) }
+        }
+        imageDatePicker.setOnClickListener { showDateTimePicker() }
+        buttonSave.setOnClickListener { saveNewTask() }
+    }
+
+    private fun onColorPicker(tag: String) {
+        taskColor = tag
```
Comment (tuanpa-0908) [inline app/src/main/java/com/sunasterisk/getitdone/ui/newTask/NewTaskFragment.kt:95]: Đặt tên thế này dễ nhầm lẫn nhỉ, gán tag(string) cho màu của task???

File: app/src/main/res/layout/fragment_new_task.xml [hunk 2]
```diff
@@ -62,13 +62,14 @@
...
         android:src="@drawable/ic_date"
-        android:tint="@android:color/holo_blue_dark"
+        android:tint="@android:color/darker_gray"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/textDatePicker"
         tools:ignore="ContentDescription" />
 
     <LinearLayout
+        android:id="@+id/linearCardView"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/fragment_new_task.xml:72]: Bậy rồi, Linear chứ có phải CardView đâu :v

File: app/src/main/java/com/sunasterisk/getitdone/ui/newTask/NewTaskFragment.kt (hunk 1 above)
Comment (quockhanhng) [reply app/src/main/java/com/sunasterisk/getitdone/ui/newTask/NewTaskFragment.kt:95]: thuộc tính color của Task là String mã hex của màu ạ

File: app/src/main/res/layout/fragment_new_task.xml (hunk 2 above)
Comment (quockhanhng) [reply app/src/main/res/layout/fragment_new_task.xml:72]: nó là layout chứa 1 list các CardView ạ :smile:

File: app/src/main/java/com/sunasterisk/getitdone/ui/newTask/NewTaskFragment.kt [hunk 3]
```diff
@@ -20,24 +43,91 @@ class NewTaskFragment : BaseBottomSheetFragment<NewTaskContract.View, NewTaskPre
...
+            presenter = NewTaskPresenter(taskRepository).apply {
+                attach(this@NewTaskFragment)
+            }
+        }
+    }
+
+    private fun initListener() {
+        for (item in linearCardView.children) {
+            if (item is CardView) item.setOnClickListener { onColorPicker(item.tag as String) }
+        }
```
Comment (aug16vcc) [inline app/src/main/java/com/sunasterisk/getitdone/ui/newTask/NewTaskFragment.kt]:
```suggestion
        linearCardView.children.filterIsInstance(CardView::class.java).forEach {
            it.setOnClickListener { onColorPicker(it.tag as String) }
        }
```

**Review by aug16vcc — APPROVED** [review]

**Pull request merged**